src/troubleshooting/content.md

```diff
@@ -0,0 +1,641 @@
+**TL;DR:** here's a diagram to help you debug your deployments in Kubernetes.
+
+![Flow chart to debug deployments in Kubernetes](diagram.png)
+
+When you wish to deploy an application in Kubernetes, you usually define three components:
+
+- a Deployment — which is a recipe for creating copies of your application called Pods
+- a Service — an internal load balancer that routes the traffic to Pods
+- an Ingress — a description of how the traffic should flow from outside the cluster to your Service.
```

Comment:
I would mention here somehow that this is only needed for end-user services and not for internal microservices.  "(optionally) an Ingress" or something like that.

Reply:
I think the entire article has an implicit assumption that you're deploying a web app with deployment+service+ingress which is an oversimplification.
But since this is >80% of the application deployed in k8s, I'm tempted to leave it like this.
Also, the sentence starts with "you *usually* define three components" because it's not always true.